Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -0,0 +1,1291 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities;
+using Microsoft.MixedReality.Toolkit.Core.EventDatum.Input;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.Core.Services;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX
+{
+    public class BoundingBox : MonoBehaviour, IMixedRealityPointerHandler, IMixedRealityGestureHandler, IMixedRealitySpatialInputHandler, IMixedRealitySourceStateHandler
+    {
+        #region Enums
+
+        private enum FlattenModeType
+        {
+            DoNotFlatten = 0,
+            /// <summary>
+            /// Flatten the X axis
+            /// </summary>
+            FlattenX,
+            /// <summary>
+            /// Flatten the Y axis
+            /// </summary>
+            FlattenY,
+            /// <summary>
+            /// Flatten the Z axis
+            /// </summary>
+            FlattenZ,
+            /// <summary>
+            /// Flatten the smallest relative axis if it falls below threshold
+            /// </summary>
+            FlattenAuto,
+        }
+
+        private enum HandleType
+        {
+            None = 0,
+            Rotation,
+            Scale
+        }
+
+        private enum WireframeType
+        {
+            Cubic = 0,
+            Cylindrical
+        }
+
+        private enum CardinalAxisType
+        {
+            X = 0,
+            Y,
+            Z
+        }
+
+        private enum BoundsCalculationMethod
+        {
+            Collider = 0,
+            Colliders,
+            Renderers,
+            MeshFilters
+        }
+
+        private enum HandleMoveType
+        {
+            Ray = 0,
+            Point
+        }
+
+        #endregion Enums
+
+        #region Serialized Fields
+
+        [Header("Bounds Calculation")]
+        [Tooltip("For complex objects, automatic bounds calculation may not behave as expected. Use an existing Box Collider (even on a child object) to manually determine bounds of Bounding Box.")]
+        [SerializeField]
+        private BoxCollider boxColliderToUse = null;
+
+        [Header("Behavior")]
+        [SerializeField]
+        private bool activateOnStart = false;
+
+        [SerializeField]
+        private float scaleMaximum = 2.0f;
+
+        [SerializeField]
+        private float scaleMinimum = 0.2f;
+
+        [Header("Wireframe")]
+        [SerializeField]
+        private bool wireframeOnly = false;
+
+        public bool WireframeOnly
```

Comment:
please add summary tag for generating api documentation

Reply:
👍 done